app/helpers/application_helper.rb

```diff
@@ -1,2 +1,10 @@
 module ApplicationHelper
+  def full_title page_title = ""
+    base_title = I18n.t :title, scope: "layouts.application", default: "Ruby on Rails Tutorial Sample App"
+    if page_title.empty?
+      base_title
+    else
+      page_title + " | " + base_title
+    end
```

Comment:
dùng toán tử 3 ngôi
so sánh nil?, empty?, blank?, present?
nên dùng cái nào trong trường hợp này

Reply:
empty

Reply:
e trả lời so sánh những method nil?, empty?, blank?, present? cụ thể vào đây luôn nhé

Reply:
nil và empty là hàm của ruby 
blank và present là hàm của rails 

- nil? check xem object có nil hay không
- empty?  check xem object length có bằng 0 hay không
- blank? trả về true đối với string k có gì hoặc string chứa toàn dấu cách hoặc array, hash, set mà k có phần tử( tương tụ empty)
- present? là hàm ngược lại với hàm blank? (!"".blank? == "".present?) 
trường hợp này nên chọn blank?